server/shared/src/main/scala/org/http4s/server/middleware/Timeout.scala

```diff
@@ -31,6 +31,15 @@ object Timeout {
     * duration if the service has not yet responded.  If the timeout
     * fires, the service's response is canceled.
     *
+    * Note, if effects of service responding is uncancelable
+    * (e.g. if service uses `MonadCancel#uncancelable` under the hood) and has exceeded
+    * the timeout, then an expected behavior is:
+    * <ul>
+    *   <li> effects will be completed naturally (regardless of how long it takes), </li>
```

Comment:
I think there is some nuance here, because cancelable and uncancelable effects can be interleaved.

```suggestion
    *   <li> uncancelable effects will be completed naturally (regardless of how long it takes), </li>
```

Reply:
So the whole thing will look like this:

> Note, if the service runs uncancelable effects while responding (e.g. if service uses `MonadCancel#uncancelable` under the hood) and has exceeded the timeout, then the expected behavior is:
> - uncancelable effects will be completed naturally (regardless of how long it takes),
> - after that, the timeout response will be returned.

I think I like it.